implementations/rust/ockam/ockam_command/src/vault/attach_key.rs

```diff
@@ -42,14 +42,13 @@ async fn run_impl(opts: CommandGlobalOpts, cmd: AttachKeyCommand) -> crate::Resu
     }
     let vault = v_state.get().await?;
     let idt = {
-        let attrs = SecretAttributes::new(SecretType::NistP256, SecretPersistence::Persistent, 32);
-        let kid = vault.secret_import(Secret::Aws(cmd.key_id), attrs).await?;
-        let attrs = KeyAttributes::new(IdentityChangeConstants::ROOT_LABEL.to_string(), attrs);
+        let attrs = SecretAttributes::NistP256;
+        let key_attrs = KeyAttributes::new(IdentityChangeConstants::ROOT_LABEL.to_string(), attrs);
         opts.state
             .get_identities(vault)
             .await?
             .identities_creation()
-            .create_identity_with_external_key(&kid, attrs)
+            .create_identity_with_external_key(&cmd.key_id, key_attrs)
```

Comment:
this was a bit entangled and too much specific.     Previously it was doing *two* unrelated things:
1)  Attach an _existing_  kms key to a kms vault  to be able to use it from ockam (that was needed so far)
2) Create an identity using _that_  keyid 

These things were not mean to be in a single command, for example there was not way for creating an identity from an existing key on a file vault,  nor to use a kms key for other thing than to create an identity from it.     These should had been  two different commands  (I would swear there was an issue for that but coudln't find it). 

But If I'm understanding well the changes here,   the step 1) here is no longer needed?   you can refer to the existing kms key  directly without "attaching it"  first?    If that's the case I'd  suggest fixing the docs for the `key_id` parameter  (this command should work for any vault type now?). 

Then, the entire `attach_key`  command make no sense anymore :),  `attach_key`  is not attaching a key is creating an identity  (this can be done on a separate PR if you don't want to include further changes here)   so  this should be just an  optional  `key_id`  parameter to the `identity create` command , indicating to use that key instead of generating a random one.

Reply:
Good catch, you're absolutely right. I am going to fix this

Reply:
You are right, let's address this in the next PR.